src/CoreLib/DSetCSharpLinq.fs

```diff
@@ -0,0 +1,104 @@
+(*---------------------------------------------------------------------------
+    Copyright 2013 Microsoft
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.                                                      
+
+    File: 
+        DSetCSharpLinq.fs
+  
+    Description: 
+        CSharp Linq APIs for DSet<'U>
+
+    Author:																	
+        Jin Li, Principal Researcher
+        Microsoft Research, One Microsoft Way
+        Email: jinl at microsoft dot com
+    Date:
+        Sept. 2015
+    
+ ---------------------------------------------------------------------------*)
+namespace Prajna.Api.CSharp.Linq
```

Comment:
So we have APIs only for FSharp, and have APIs that is for general .Net (thus can be used by both F#, C#, or VB).  However, the latter is assumed to be used mainly by C# user. So currently the structure is:
* Prajna.Core:  runtime APIs (e.g. Cluster) that can be used from both C#/F#
* Prajna.Api.FSharp: F# only APIs
* Prajna.Api.CSharp: C# APIs, (but in fact usable in any .Net language)
Prajna.Tools is structured in a similar way. I think we can further discuss the choice here. But this pull request would not address it, since it's a much larger scope of decision.